tests/sentry/snuba/test_discover.py

```diff
@@ -0,0 +1,390 @@
+                "count_unique(transaction.duration)",
+            ],
+            query="",
+            params={"project_id": [self.project.id]},
+        )
+        mock_query.assert_called_with(
+            selected_columns=["transaction", "duration"],
+            aggregations=[
+                ["uniq", "duration", "count_unique_transaction_duration"],
+                ["argMax", ["event_id", "timestamp"], "latest_event"],
+                ["argMax", ["project_id", "timestamp"], "projectid"],
+            ],
+            filter_keys={"project_id": [self.project.id]},
+            dataset=Dataset.Discover,
+            end=None,
+            start=None,
+            conditions=[],
+            groupby=["transaction", "duration"],
+            orderby=None,
+            referrer=None,
+        )
+
+    @patch("sentry.snuba.discover.raw_query")
+    def test_selected_columns_aggregate_alias(self, mock_query):
+        mock_query.return_value = {
+            "meta": [{"name": "transaction"}, {"name": "p95"}],
+            "data": [{"transaction": "api.do_things", "p95": 200}],
+        }
+        discover.query(
+            selected_columns=["transaction", "p95", "count_unique(transaction)"],
+            query="",
+            params={"project_id": [self.project.id]},
+        )
+        mock_query.assert_called_with(
+            selected_columns=["transaction"],
+            aggregations=[
+                ["quantile(0.95)(duration)", "", "p95"],
+                ["uniq", "transaction", "count_unique_transaction"],
+                ["argMax", ["event_id", "timestamp"], "latest_event"],
+                ["argMax", ["project_id", "timestamp"], "projectid"],
+            ],
+            filter_keys={"project_id": [self.project.id]},
+            dataset=Dataset.Discover,
+            groupby=["transaction"],
+            conditions=[],
+            end=None,
+            start=None,
+            orderby=None,
+            referrer=None,
+        )
+
+    @patch("sentry.snuba.discover.raw_query")
+    def test_orderby(self, mock_query):
+        mock_query.return_value = {
+            "meta": [{"name": "title"}, {"name": "project.id"}],
+            "data": [{"project.id": "tester", "title": "test title"}],
+        }
+        discover.query(
+            selected_columns=["project.id", "title"],
+            query="",
+            params={"project_id": [self.project.id]},
+            orderby=["project.id"],
+        )
+        mock_query.assert_called_with(
+            selected_columns=["project_id", "title", "event_id"],
+            filter_keys={"project_id": [self.project.id]},
+            dataset=Dataset.Discover,
+            orderby=["project_id"],
+            aggregations=[],
+            end=None,
+            start=None,
+            conditions=[],
+            groupby=[],
+            referrer=None,
+        )
+
+    @patch("sentry.snuba.discover.raw_query")
+    def test_orderby_must_be_selected(self, mock_query):
```

Comment:
Why does orderby have to be selected? It's not a requirement of Snuba. Discover 1 (and I believe various other parts of the app) rely on using orderby terms that are not being selected.

Reply:
I've made it a requirement so that people don't attempt to sort the result set based on aggregates that are undefined. For example selecting `project.id, title` and ordering but `-count(id)` is not currently accepted because it requires implicit aggregation.